### PR TITLE
Actions improvement and UX improvement when Actions is running

### LINKED ISSUE #156: Add the delete Actions history button
Add the delete Actions history button to make the Actions page more clean.

### LINKED ISSUE #157: Make admin cannot run the Actions script when same Actions is running
To avoid running the same Action, we must need to block on running the same Action to avoid the conflict and not make the server run too harsh.

### LINKED ISSUE #158: Fire a message on other page when the Actions is running
Make the message to show when the Action is running so when on update progress if user see something wrong on card and see this message user know that this is not a bug.

### PR DESCRIPTION
**Pull request by HelloYeew**
Hello! This is the prull request to update our Actions or our data update system.

# Fire a message when the update Action is running

Close #158 

image

When the system detect that the Actions is running on updating the beatmap metadata system, the system will notified user so if user feel interruption on the beatmap card the user will now know that it's because the update system. I think the message can be better...

# Make admin cannot run the Actions script when same Actions is running

Close #157 

image

To make the server is not loading too much due to a lot of Actions, if the system detect that it has the Actions that is running with the same name it will block from starting the new worker with a same Actions scheme.

I also adding a block on the launch view to avoid user running it via the view or URL that we set.

# Add the delete Actions log button

Close #156 

image

Add the delete log button to avoid the Actions page not have too much log on one page. As before, the button will update real-time when the status of the worker is change.

## Security method

I have added to protect the log deletion when that log is bind with the worker that is running or in the idle state to avoid the error.

# Other improvement

- Fix method to fetch log url cannot handle error when no log file
- Delete unused code
- Add error handle when the action finish but cannot calculate time due to no finish time

Please try on it before approve. Thanks!

### CODE REVIEW

File: actions/templates/actions/actions.html [hunk 1]
```diff
@@ -101,6 +108,12 @@ <h3><b>#{{ action.id }}</b> {{ action.name }}</h3>
     {% if action.get_log_url %}
     <p><a class="text-decoration-none text-center spacing-hover-short hvr-icon-bounce" href="{{ action.log.url }}"><i class="fas fa-file-download icon-menu hvr-icon"></i> Download log</a></p>
     {% endif %}
+
+    {% if action.status == 2 or action.status == 3 %}
```
Comment (pontakornth) [inline actions/templates/actions/actions.html:112]: Can you explain what is status 2 and 3?

File: actions/templates/actions/actions.html [hunk 2]
```diff
@@ -101,6 +108,12 @@ <h3><b>#{{ action.id }}</b> {{ action.name }}</h3>
...
+    <a class="btn btn-danger hvr-sweep-to-right-danger" id="delete_log_{{ action.id }}" href="{% url 'delete_action_log' action.id %}" role="button"><i class="fas fa-trash"></i> Delete Log</a>
+    {% else %}
+    <a class="btn btn-danger disabled hvr-sweep-to-right-danger" id="delete_log_{{ action.id }}" href="{% url 'delete_action_log' action.id %}" role="button" aria-disabled="true"><i class="fas fa-trash"></i> Delete Log</a>
```
Comment (pontakornth) [inline actions/templates/actions/actions.html:115]: Bootstrap link does not disable keyboard.

**Review by pontakornth — CHANGES_REQUESTED** [review]

File: actions/templates/actions/actions.html (hunk 1 above)
Comment (HelloYeew) [reply actions/templates/actions/actions.html:112]: The status 2 mean the actions is complete and status 3 is status is failed.

File: actions/templates/actions/actions.html (hunk 2 above)
Comment (HelloYeew) [reply actions/templates/actions/actions.html:115]: That's mean we need to change the 'href' link when the button is disable to?

File: actions/templates/actions/actions.html [hunk 3]
```diff
@@ -57,10 +57,17 @@ <h1 data-aos="fade-up" data-aos-duration="700" data-aos-once="true"><i class="fa
     </div>
     <div class="row">
         <div class="col-4" data-aos="fade-up" data-aos-duration="850" data-aos-once="true">
-          <a class="btn btn-warning hvr-bounce-to-right-warning" href="{% url 'actions_update_beatmap' %}">Update all beatmaps metadata</a>
+            {% if update_beatmap_running %}
+            <a class="btn btn-warning hvr-bounce-to-right-warning disabled" href="๒" aria-disabled="true">Update all beatmaps metadata</a>
```
Comment (pontakornth) [inline actions/templates/actions/actions.html:61]:
```suggestion
            <a class="btn btn-warning hvr-bounce-to-right-warning disabled" href="#" aria-disabled="true">Update all beatmaps metadata</a>
```
Misspelling.

File: actions/views.py [hunk 4]
```diff
@@ -45,13 +53,19 @@ def check_action_log(request, log_id):
...
+    This view can only access by superuser and staff.
     """
     action = get_object_or_404(ActionLog, id=log_id)
-    if action.status == 1 or action.status == 0:
-        duration = (timezone.now() - action.time_start).total_seconds()
-    elif action.status == 2:
-        duration = (action.time_finish - action.time_start).total_seconds()
-    else:
+    try:
+        if action.status == 1 or action.status == 0:
```
Comment (pontakornth) [inline actions/views.py:61]: Can you add some comments or docstring in this code to explain the status number?

**Review by pontakornth — CHANGES_REQUESTED** [review]

File: actions/views.py (hunk 4 above)
Comment (HelloYeew) [reply actions/views.py:64]: I've added it. Sorry for that.

**Review by pontakornth — APPROVED** [review]

**Review by some-random-idiot — APPROVED** [review]

**Review by SainTurDaY27 — APPROVED** [review]
Nice one!

**Review by Siraphop4Nene — APPROVED** [review]

**Comment by HelloYeew** [comment]
Let's merge it!

**Pull request merged**